src/conversion.jl

```diff
@@ -1,3 +1,22 @@
+"""
+    UnitConversionFactor(x::T) where {T<:AbstractFloat}
+Conversion factor with value `x`.
+
+Used by the [`convfact`](@ref) function to preserve
+the floating-point precision of quantities.
+"""
+struct UnitConversionFactor{T<:AbstractFloat} <: AbstractIrrational
+    x::T
+    # the inner constructor necessary for ambiguity resolution
+    UnitConversionFactor(x::T) where {T<:AbstractFloat} = new{T}(x) 
+end
+
```

Comment:
```suggestion

Base.:*(a::UnitConversionFactor, b::BigFloat) = a.x * b
Base.:*(a::BigFloat, b::UnitConversionFactor) = a * b.x
```
When multiplying a `BigFloat` by a `UnitConversionFactor{Float64}`, the conversion factor would be converted to `BigFloat`, so it would allocate a new `BigFloat` without gaining any precision. This can be avoided by adding these two methods. This fixes a slight performance regression:
```julia
julia> @btime uconvert(u"rad", $(big(1.0)u"°"));
  111.622 ns (2 allocations: 104 bytes) # v1.21.1
  125.055 ns (4 allocations: 208 bytes) # this PR (without the suggested change)

julia> @btime uconvert(u"rad", $(big(1.0+2.0im)u"°"));
  221.463 ns (4 allocations: 208 bytes) # v1.21.1
  251.567 ns (8 allocations: 416 bytes) # this PR (without the suggested change)
```